elements/pl-code/pl-code.py

```diff
@@ -178,7 +212,7 @@ def render(element_html, data):
             base_path = os.path.join(data["options"]["question_path"], directory)
         file_path = os.path.join(base_path, source_file_name)
         if not os.path.exists(file_path):
-            raise Exception(f'Unknown file path: "{file_path}".')
+            raise ValueError(f'Unknown file path: "{file_path}".')
```

Comment:
Somewhat tangential (but we're already rearranging this file): two lines below this, I think the ending newline removal is broken, and the file doesn't have to be manually closed.

Reply:
Replaced with builtins in 17aa3b722.

Reply:
Are the Canadian servers on Python >= 3.9? Those builtins are from Python 3.9, but if no versioning issues I think the new code looks much cleaner?

Reply:
3.10.6 https://prairielearn.slack.com/archives/C0XBTPZMZ/p1673396644523079?thread_ts=1673392551.982729&cid=C0XBTPZMZ

Reply:
It seems you're on top of it, but apparently pyright doesn't support it yet...

```
pyright 1.1.295
/home/jonatan/pl/elements/pl-code/pl-code.py
  /home/jonatan/pl/elements/pl-code/pl-code.py:221:29 - error: Cannot access member "removesuffix" for type "str"
    Member "removesuffix" is unknown (reportGeneralTypeIssues)
  /home/jonatan/pl/elements/pl-code/pl-code.py:235:39 - error: Cannot access member "removeprefix" for type "str"
    Member "removeprefix" is unknown (reportGeneralTypeIssues)
```

Reply:
Yeah, no idea why pyright doesn't pick that up, weird.